packages/widget/src/context/wallet.ts

```diff
@@ -54,14 +70,76 @@ export class WalletContextProvider extends BaseComponent {
   @provide({ context: walletContext })
   private walletContext: WalletContext = {};
 
+  @provide({ context: substrateProviderContext })
+  substrateProviderContext: SubstrateProviderContext = {};
+
   @property({ attribute: false, type: Object })
   evmWallet?: EvmWallet;
 
-  connectedCallback(): void {
+  @property({ attribute: false })
+  substrateProviders?: Array<ApiPromise> = [];
+
+  @property({ type: String }) environment?: Environment;
+
+  get defaultSubstrateProviders(): Array<ApiPromise> {
```

Comment:
maybe a nice solution would be instead of SUBSTRATE_CHAIN_RPCS url's to have parachainId -> ApiPromises direct mappings to avoid useless rpc requests 🤷🏻‍♂️

Reply:
@mpetrunic if we do that, we will make additional call even if substrate providers were provided by the third party. We dont need to have api promise objects if user have defined them.

Reply:
`defaultSubstrateProviders` is only used when `substrateProviders` are not provided by the 3rd party

Reply:
`defaultProviders` only makes calls when nothing is specified by user 🤔

Reply:
But why? Wouldn't it make more sense to always have defaults unless user overrides some or adds new ones?

Reply:
@mpetrunic it does, and that's how it is right now. having ws urls hardcoded as strings is better than having default provider objects. if user overrides or provides theirs through property `substrateProviders` current logics pick them over the default rpc urls and no object is created.

What you are suggesting will always have provider objects regardless of user specifying them or not

Reply:
Is this now more clear in regards to: https://github.com/sygmaprotocol/sygma-widget/pull/134#discussion_r1557630207 ? 😅

Reply:
@mpetrunic yes it is! thank you very much